standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStoreChecker.java

```diff
@@ -356,7 +347,7 @@ void checkTable(Table table, PartitionIterable parts, byte[] filterExp, CheckRes
       }
     }
 
-    findUnknownPartitions(table, partPaths, filterExp, result);
+    findUnknownPartitions(table, partPaths, partPathsInMS, filterExp, result);
```

Comment:
Am I right when I think that the `partPaths` and the `partPathsInMS` contain basically the same path objects.
If the dir is like this
```
---root
 + -+---P=1/K=1
  |    \---P=1/K=2
  +-+---P=3/K=1
       \---P=3/K=3
```
`partPathsInMS`:
- root/P=1/K=1
- root/P=1/K=2
- root/P=3/K=1
- root/P=3/K=3

`partPath`:
- root
- root/P=1
- root/P=1/K=1
- root/P=1/K=2
- root/P=3
- root/P=3/K=1
- root/P=3/K=3

Do I understand the above correctly?

Do we need to send all of these to the filter? Isn't this duplicated info?